src/Models/incompressible_model.jl

```diff
@@ -74,15 +73,15 @@ function IncompressibleModel(;
                coriolis = nothing,
           surface_waves = nothing,
                 forcing = ModelForcing(),
-    boundary_conditions = SolutionBoundaryConditions(grid),
+    boundary_conditions = NamedTuple(),
```

Comment:
The grid is required `kwarg`, which means that we can at least infer default velocity boundary conditions here. Perhaps that would be better?

Reply:
Sorry I should have been clearer what the model constructor does. Passing the empty `NamedTuple()` will impose default boundary conditions on every field as the `boundary_conditions` are passed to all the field tuple constructors.

Users now impose non-default boundary conditions by passing a named tuple of `FieldBoundaryConditions` where the keys are field names.

Some examples:
```julia
boundary_conditions = (u=u_bcs,)
boundary_conditions = (u=u_bcs, T=T_bcs)
boundary_conditions = (u=u_bcs, T=T_bcs, κₑ=(T=κₑ_bcs, S=κₑ_bcs))
```

An alternative API design could be a more hierarchical but more verbose named tuple (although I'm not a super huge fan of the verbosity here unless you want to call your tracers u, v, w, or κₑ for some reason):
```julia
boundary_conditions = (velocities=(u=u_bcs,),)
boundary_conditions = (velocities=(u=u_bcs,), tracers=(T=T_bcs,))
boundary_conditions = (velocities=(u=u_bcs,), tracers=(T=T_bcs,), diffusivities=(T=κₑ_bcs, S=κₑ_bcs))
```

Reply:
Yeah, I guess either way is fine. We could just write

```julia
boundary_conditions = (u = UVelocityBoundaryConditions(grid), v=VVelocityBoundaryConditions(grid), w=WVelocityBoundaryConditions(grid))
```

The only real purpose of doing that is code-as-documentation (if you're reading the model constructor, you can use the default as an example for what you need to do). Not sure if that's a good practice or not.

Reply:
> Of course if non-defaults are used for some fields then we still have to infer defaults for the rest of the fields so we don't really help ourselves practically with a default.

True but the examples and model setup docs have plenty of examples of non-default boundary conditions so I don't think it's a bad idea to show an actual named tuple in the docstring as opposed to an empty `NamedTuple()` which you'd almost never use as a user.